tests/unit_tests/data_steward/gcloud/bq/bq_test.py

```diff
@@ -40,6 +45,109 @@ def _get_all_field_types(self,) -> typing.FrozenSet[str]:
+        :param key_order: If `rows` refers to a list of dict rather than OrderedDict, 
+            specifies how fields are ordered in the result schema. This parameter is 
+            ignored if `rows` refers to a list of OrderedDict.
+        :return: a mock RowIterator
+        """
+        mock_row_iter = MagicMock(spec=bigquery.table.RowIterator)
+        mock_row_iter.total_rows = len(rows)
+        row0 = rows[0]
+        if isinstance(row0, OrderedDict):
+            _rows = rows
+        else:
+            _rows = [self._to_ordered_dict(row, key_order) for row in rows]
+            row0 = _rows[0]
+
+        mock_row_iter.schema = list(self.fields_from_dict(row0))
+        field_to_index = {key: i for i, key in enumerate(row0.keys())}
+        mock_row_iter.__iter__ = Mock(return_value=iter([
+            bigquery.table.Row(list(row.values()), field_to_index)
+            for row in _rows
+        ]))
+        return mock_row_iter
+
+    def _to_ordered_dict(self, d: dict, key_order: Iterable[Any] = None):
+        """
+        Convert a dict to OrderedDict with a specified order
+        :param d: instance to convert
+        :param key_order: specifies how items are ordered in the result
+        :return: the ordered dict
+        """
+        if len(d) == 1:
+            return OrderedDict(d)
+        else:
+            if key_order is None:
+                raise ValueError(
+                    'Parameter key_order is required in order to convert'
+                    ' a dict with multiple items to OrderedDict')
+        return OrderedDict((key, d[key]) for key in key_order)
+
+    def fields_from_dict(self, row):
+        """
+        Get schema fields from a row represented as a dictionary
+
+        :param row: the dictionary to infer schema for
+        :return: list of schema field objects
+        
+        Example:
+            >>> from tests import bq_test_helpers
+            >>> d = {'item': 'book', 'qty': 2, 'price': 1.99}
+            >>> bq_test_helpers.fields_from_dict(d)
+                [SchemaField('item', 'STRING', 'NULLABLE', None, ()),
+                SchemaField('qty', 'INT64', 'NULLABLE', None, ()),
+                SchemaField('price', 'FLOAT64', 'NULLABLE', None, ())]
+        """
+        return [
+            self._field_from_key_value(key, value)
+            for key, value in row.items()
+        ]
+
+    def _field_from_key_value(self, key: str,
+                              value: Any) -> bigquery.SchemaField:
+        """
+        Get a schema field object from a key and value
+        
+        :param key: name of the field
+        :param value: value of the field
+        :return: an appropriate schema field object
+        """
+        _TYPE_TO_FIELD_TYPE = {
+            str(int): 'INT64',
+            str(str): 'STRING',
+            str(float): 'FLOAT64',
+            str(datetime.date): 'DATE',
+            str(datetime.datetime): 'TIMESTAMP'
+        }
```

Comment:
This mapping would be great to have in a constants file. Maybe as Python type to BigQuery type to be more specific?

Reply:
I thought I would put it inside `def _field_from_key_value` since it's used only there. Maybe relocate it in future if there's more use for it?

Reply:
Yes, that works too. I think there are other usages but they have their own dictionary. Maybe it can be consolidated in a separate ticket.